modules/core/src/test/java/org/apache/ignite/internal/processors/cache/checker/processor/PartitionReconciliationAtomicLongDataStructureTest.java

```diff
@@ -0,0 +1,180 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.apache.ignite.internal.processors.cache.checker.processor;
+
+import java.lang.reflect.Constructor;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import org.apache.ignite.configuration.AtomicConfiguration;
+import org.apache.ignite.configuration.DataRegionConfiguration;
+import org.apache.ignite.configuration.DataStorageConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.IgniteEx;
+import org.apache.ignite.internal.processors.cache.GridCacheContext;
+import org.apache.ignite.internal.processors.cache.checker.objects.ReconciliationResult;
+import org.apache.ignite.internal.processors.cache.verify.PartitionReconciliationKeyMeta;
+import org.apache.ignite.internal.processors.cache.verify.RepairAlgorithm;
+import org.apache.ignite.internal.processors.datastructures.GridCacheInternalKeyImpl;
+import org.apache.ignite.testframework.junits.WithSystemProperty;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import static org.apache.ignite.IgniteSystemProperties.IGNITE_SENSITIVE_DATA_LOGGING;
+
+/**
+ * Tests that reconciliation works with data structures.
+ */
+@RunWith(Parameterized.class)
+public class PartitionReconciliationAtomicLongDataStructureTest extends PartitionReconciliationAbstractTest {
+    /** Nodes. */
+    protected static final int NODES_CNT = 4;
+
+    /** Keys count. */
+    protected static final int KEYS_CNT = 100;
+
+    /** Corrupted keys count. */
+    protected static final int BROKEN_KEYS_CNT = 10;
+
+    /** Data structure name. */
+    protected static final String DS_NAME = "DefaultDS";
+
+    /** Fix mode. */
+    @Parameterized.Parameter(0)
+    public boolean fixMode;
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String name) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(name);
+
+        cfg.setDataStorageConfiguration(new DataStorageConfiguration()
+            .setDefaultDataRegionConfiguration(new DataRegionConfiguration()
+                .setPersistenceEnabled(true)
+                .setMaxSize(300L * 1024 * 1024))
+        );
+
+        cfg.setConsistentId(name);
+        cfg.setAutoActivationEnabled(false);
+
+        return cfg;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTest() throws Exception {
+        stopAllGrids();
+
+        cleanPersistenceDir();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        stopAllGrids();
+
+        cleanPersistenceDir();
+    }
```

Comment:
I think nowadays you're supposed to use `@After` instead of overriding afterTest. Also, if you're overriding you need to call `super.afterTest()` at the end.

Reply:
I decided to keep the test in sync with other ones. Probably we can refactor all of them in a package separately.